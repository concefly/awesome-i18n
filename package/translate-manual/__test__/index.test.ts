import { TranslateCtx, TranslateItem } from 'ai18n-type';
import ManualTranslator from '../lib';

class MockManualTranslator extends ManualTranslator {
  answerList: any[] = [];
  fileData = new Map<string, string>();

  protected answerIndex = 0;
  protected prompts = (async () => {
    return this.answerList[this.answerIndex++];
  }) as any;

  protected fs = {
    writeFileSync: (path: string, data: any) => this.fileData.set(path, data),
    readFileSync: (path: string) => this.fileData.get(path),
    unlinkSync: (path: string) => this.fileData.delete(path),
  } as any;
}

const createCtx = () => {
  return new TranslateCtx([
    new TranslateItem('苹果', 'zh-cn', 'en'),
    new TranslateItem('香蕉', 'zh-cn', 'en'),
  ]);
};

const inspectCtx = (ctx: TranslateCtx) => {
  return ctx;
};

describe('ManualTranslator', () => {
  it('translateManual', async () => {
    const trans = new MockManualTranslator();
    trans.answerList = [{ transType: 'manual' }, { message: 'apple' }, { message: 'banana' }];

    const ctx = createCtx();
    await trans.translate(ctx);

    expect(inspectCtx(ctx)).toMatchSnapshot();
  });

  it('translatePlatform', async () => {
    const trans = new MockManualTranslator();
    trans.answerList = [{ transType: 'platform' }, { confirm: true }];

    const ctx = createCtx();
    await trans.translate(ctx);

    expect(trans.fileData.has('translate-manual.txt')).toBeFalsy();
    expect(inspectCtx(ctx)).toMatchSnapshot();
  });
});
