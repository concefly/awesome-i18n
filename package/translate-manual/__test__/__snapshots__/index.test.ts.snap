// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManualTranslator translateManual 1`] = `
TranslateCtx {
  "list": Array [
    TranslateItem {
      "from": "zh-cn",
      "result": Object {
        "message": "apple",
      },
      "text": "苹果",
      "to": "en",
      "uglify": Object {
        "mapper": Map {},
        "text": "苹果",
      },
    },
    TranslateItem {
      "from": "zh-cn",
      "result": Object {
        "message": "banana",
      },
      "text": "香蕉",
      "to": "en",
      "uglify": Object {
        "mapper": Map {},
        "text": "香蕉",
      },
    },
  ],
}
`;

exports[`ManualTranslator translatePlatform 1`] = `
TranslateCtx {
  "list": Array [
    TranslateItem {
      "from": "zh-cn",
      "result": Object {
        "message": "苹果",
      },
      "text": "苹果",
      "to": "en",
      "uglify": Object {
        "mapper": Map {},
        "text": "苹果",
      },
    },
    TranslateItem {
      "from": "zh-cn",
      "result": Object {
        "message": "香蕉",
      },
      "text": "香蕉",
      "to": "en",
      "uglify": Object {
        "mapper": Map {},
        "text": "香蕉",
      },
    },
  ],
}
`;
